### DISCUSSION TITLE
Merging this project with react-qr-reader

### DISCUSSION THREAD

**Discussion by JodusNodus** [BlackBoxVision/react-qr-reader#51] [Ideas]
Hey, i've recently been looking into getting react-qr-reader refactored so that it will work with modern stacks. Because as you know i've abandoned it for quite a long time. I've been looking into a refactor with typescript and rollup. I see that you guys have made a lot of progress already. I was wondering if you'd be interested in merging these changes into the official project and maintaining it.

**Comment by JonatanSalas** [comment]
Hi @JodusNodus, first of all, thanks for your project which has helps us a lot when building web applications that require QR Code Scanning capabilities. 

We're open to work together in order to merge our changes into the core repository. How do you want us to handle this? 

Also, since this changes would be in the core repo which is on your user, I would like to ask you obviously if you want, could we create an organization for `react-qr-reader` so we can work on building also a community?

**Comment by JodusNodus** [comment]
Hey, the main problem is that i no longer use the project in my job or personal projects anymore. So that explain's the lack of motivation to follow up on everything. Which is where you guys could help me out. I question if the project is big enough for a seperate organisation. if alright with you, you can make a pull request. Let me if i should add someone else as a maintainer.

Technical question, why did you choose to replace jsqr with zxing, back when i first build it jsqr was quicker and using it in a web worker meant that the main thread could render react without any lag.

**Comment by JonatanSalas** [comment]
I propose an organization because in our internal roadmap we've the intention to build this library as a universal library (react and react native/expo support), so we think it will scale a little up. 

About the technical question, we've replaced `jsQR` due to issues with specific QR types, like Mini QRs. Seems jsQR isn't spec compliant on the scanning support. We moved forward to zxing because its more spec compliant. 

About using a web worker, that's true, we've the intention to integrate the support again because it would make the library perform better on low range devices. 

We're looking forward to use this library: https://github.com/alewin/useWorker

**Comment by JodusNodus** [comment]
Gotcha, that makes sense. I've setup the org and invited you. The repo has been transfered. Ideally, push your local changes into a new branch on the repo. Then i can complete the setup. I'd like to move the qr processing back into a worker, so ill look into that this week.

**Reply by JonatanSalas** [reply]
Great, I'm now on the repo, hey, would you mind to enable project management, so we can make a board and I can create issues to track the progress of a newer release. I would like to give visibility to the people that's looking around the project.

**Reply by JodusNodus** [reply]
Sounds good, done.

**Reply by JonatanSalas** [reply]
Thank you!
